pkg/versioning/properties.go

```diff
@@ -0,0 +1,90 @@
+package versioning
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/magiconair/properties"
+	"github.com/pkg/errors"
+)
+
+// PropertiesFile defines an artifact using a properties file for versioning
+type PropertiesFile struct {
+	path             string
+	content          *properties.Properties
+	versioningScheme string
+	versionField     string
+	writeFile        func(string, []byte, os.FileMode) error
+}
+
+func (p *PropertiesFile) init() error {
+	if len(p.versionField) == 0 {
+		p.versionField = "version"
+	}
+	if p.writeFile == nil {
+		p.writeFile = ioutil.WriteFile
+	}
+	if p.content == nil {
+		var err error
+		p.content, err = properties.LoadFile(p.path, properties.UTF8)
```

Comment:
UTF-8 encoding may or may not be correct, no?

Reply:
we use the same in the Sonar step - other option is `ISO_8859_1`
I would suggest to go with `UTF8` for now and see if there is potentially a need for a config option further down the road